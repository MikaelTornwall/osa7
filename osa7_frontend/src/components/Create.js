import React from 'react'
import { connect } from 'react-redux'
import { notify } from '../reducers/notificationReducer'
import { create } from '../reducers/blogReducer'

export class Create extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      title: '',
      author: '',
      url: ''
    }
  }

  submit = async (event) => {
    event.preventDefault()
    console.log('New blog submitted: ', this.state.title, this.state.author, this.state.url)
    try {
      const blogObject = {
        title: this.state.title,
        author: this.state.author,
        url: this.state.url
      }

      if (blogObject.title.length === 0 || blogObject.author.length === 0 || blogObject.url.length === 0) {
        this.props.notify('All fields must be filled')
        return
      }

      this.setState({
        title: '',
        author: '',
        url: ''
      })

      this.props.create(blogObject)
      this.props.notify(`A blog "${this.state.title}" by author ${this.state.author} has been created.`)
    } catch (exception) {
      this.props.notify('All fields must be filled')
    }
  }

  handleNewBlog = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }

  render() {
    return (
      <div>
        <h3>Create new blog</h3>
        <form onSubmit={this.submit}>
          <table>
            <tbody>
              <tr>
                <td>Title</td>
                <td><input
                  type="text"
                  name="title"
                  value={this.state.title}
                  onChange={this.handleNewBlog}
                /></td>
              </tr>
              <tr>
                <td>Author</td>
                <td><input
                  type="text"
                  name="author"
                  value={this.state.author}
                  onChange={this.handleNewBlog}
                />
                </td>
              </tr>
              <tr>
                <td>Url</td>
                <td><input
                  type="url"
                  name="url"
                  value={this.state.url}
                  onChange={this.handleNewBlog}
                />
                </td>
              </tr>
              <tr>
                <td></td>
                <td><input
                  type="submit"
                  value="Create"
                />
                </td>
              </tr>
            </tbody>
          </table>
        </form>
      </div>
    )
  }
}

const mapDispatchToProps = {
  notify,
  create
}

const ConnectedCreate = connect(
  null,
  mapDispatchToProps
)(Create)

export default ConnectedCreate
