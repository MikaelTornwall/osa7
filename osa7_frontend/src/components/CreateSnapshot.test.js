import React from 'react'
import { Create } from './Create'
import renderer from 'react-test-renderer'

describe('<Create />', () => {
  it('renders correctly', () => {

    const tree = renderer
      .create(<Create />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
