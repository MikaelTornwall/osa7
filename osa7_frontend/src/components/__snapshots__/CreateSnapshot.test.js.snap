// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create /> renders correctly 1`] = `
<div>
  <h3>
    Create new blog
  </h3>
  <form
    onSubmit={[Function]}
  >
    <table>
      <tbody>
        <tr>
          <td>
            Title
          </td>
          <td>
            <input
              name="title"
              onChange={[Function]}
              type="text"
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            Author
          </td>
          <td>
            <input
              name="author"
              onChange={[Function]}
              type="text"
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            Url
          </td>
          <td>
            <input
              name="url"
              onChange={[Function]}
              type="url"
              value=""
            />
          </td>
        </tr>
        <tr>
          <td />
          <td>
            <input
              type="submit"
              value="Create"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;
